src/rgw/rgw_op.cc

```diff
@@ -5583,11 +5577,34 @@ void RGWCompleteMultipart::execute()
 			    s->bucket_info, meta_obj, 0);
   if (r < 0) {
     ldout(store->ctx(), 0) << "WARNING: failed to remove object " << meta_obj << dendl;
-    r = l.unlock(&ioctx, raw_meta_oid);
```

Comment:
if this delete succeeds, you can set locked=false to avoid sending the unlock

Reply:
ah, right